PowerUp/app/src/main/java/powerup/systers/com/ListViewMaxHeight.java

```diff
@@ -0,0 +1,40 @@
+package powerup.systers.com;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.util.AttributeSet;
+import android.widget.ListView;
+
+public class ListViewMaxHeight extends ListView {
+
+    private final int maxHeight;
+
+    public ListViewMaxHeight(Context context) {
+        this(context, null);
+    }
+
+    public ListViewMaxHeight(Context context, AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public ListViewMaxHeight(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        if (attrs != null) {
+            TypedArray styleAttributes = getContext().obtainStyledAttributes(attrs, R.styleable.ListViewMaxHeight);
+            maxHeight = styleAttributes.getDimensionPixelSize(R.styleable.ListViewMaxHeight_maxHeight, Integer.MAX_VALUE);
+            styleAttributes.recycle();
+        } else {
+            maxHeight = 0;
+        }
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        int measuredHeight = MeasureSpec.getSize(heightMeasureSpec);
+        if (maxHeight > 0 && maxHeight < measuredHeight) {
+            int measureMode = MeasureSpec.getMode(heightMeasureSpec);
+            heightMeasureSpec = MeasureSpec.makeMeasureSpec(maxHeight, measureMode);
```

Comment:
Do not assign a value to a parameter in a method. This is very error-prone. Define a new variable if needed